client/a8c-for-agencies/sections/woopayments/hooks/use-fetch-woopayments-data.tsx

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { useSelector } from 'calypso/state';
+import { getActiveAgencyId } from 'calypso/state/a8c-for-agencies/agency/selectors';
+
+export default function useFetchWooPaymentsData( {
+	startDate,
+	endDate,
+	segment,
+}: {
+	/** Start date in YYYY-MM-DD format */
+	startDate?: string;
+	/** End date in YYYY-MM-DD format */
+	endDate?: string;
+	/** Time segment: 'day' | 'week' | 'month' | 'quarter' | 'year' */
+	segment?: 'day' | 'week' | 'month' | 'quarter' | 'year';
+} ) {
+	const agencyId = useSelector( getActiveAgencyId );
+
+	const isApiEnabled = false;
+
+	return useQuery( {
+		queryKey: [ 'a4a-site-woopayments-data', agencyId, startDate, endDate, segment, isApiEnabled ],
+		queryFn: () =>
+			isApiEnabled
+				? wpcom.req.get(
+						{
+							apiNamespace: 'wpcom/v2',
+							path: `/agency/${ agencyId }/woopayments`,
+						},
+						{
+							...( startDate && { start_date: startDate } ),
+							...( endDate && { end_date: endDate } ),
+							...( segment && { segment } ),
+						}
+				  )
+				: Promise.resolve( [] ),
+		enabled: !! agencyId,
+		refetchOnWindowFocus: false,
+		staleTime: 0,
```

Comment:
I'd suggest we keep staleTime quite high for this, as it shouldn't change within given day and the query is pretty expensive. What do you think?

Reply:
Makes sense. Removed it now: https://github.com/Automattic/wp-calypso/pull/100547/commits/ed5820e586edc8e27b84dd0e234ecdf804856fcc

This is not yet being used. I'll add "one day" when we implement this.